examples/sim_fibo_ad_cp.cpp

```diff
@@ -206,6 +207,38 @@ try
                                   *props, deck, gravity[2], state);
     }
 
+    std::shared_ptr<BlackoilState> distributed_state=Dune::stackobject_to_shared_ptr(state);
```

Comment:
I am not saying that your code was buggy and introduced new lifetime errors (it looked fine to my eyes in that respect), but that the function `stackobject_to_shared_ptr` should not be used unless absolutely necessary.